guiEditor/src/components/propertyTab/propertyGrids/gui/gridPropertyGridComponent.tsx

```diff
@@ -28,27 +37,36 @@ export class GridPropertyGridComponent extends React.Component<IGridPropertyGrid
     private _columnDefinitions: string[] = [];
     private _editedRow: boolean = false;
     private _editedColumn: boolean = false;
+    private _rowChild: boolean = false;
+    private _columnChild: boolean = false;
 
     renderRows() {
-        return this._rowDefinitions.map((rd, i) => {
-            return (
-                <div key={`r${i}`} className="ge-divider">
-                    <TextInputLineComponent
-                        lockObject={this.props.lockObject}
-                        key={`rText${i}`}
-                        label={`Row ${i}`}
-                        value={rd}
-                        numbersOnly={true}
-                        onChange={(newValue) => {
-                            this._rowDefinitions[i] = newValue;
-                            this._rowEditFlags[i] = true;
-                            this._editedRow = true;
-                            this.forceUpdate();
-                        }}
-                    />
-                </div>
-            );
-        });
+        return (
+            this._rowDefinitions.map((rd, i) => {
+                return (
+                    <div key={`r${i}`} className={this._removingRow && i === this._rowEditFlags.length - 1 ? "ge-grid-remove" : this._rowEditFlags[i] ? "ge-grid-edit" : "ge-grid"}>
+                        <div className="ge-grid-divider">
+                            <TextInputLineComponent lockObject={this.props.lockObject} key={`rText${i}`} label="" icon={gridColumnIconDark} iconLabel={`Row ${i}`} value={rd} numbersOnly={true}
+                                onChange={(newValue) => {
+                                    this._rowDefinitions[i] = newValue;
+                                    this._rowEditFlags[i] = true;
+                                    this._editedRow = true;
+                                    this._removingRow = false;
+                                    this._rowChild = false;
+                                    this.forceUpdate();
+                                }} />
+                            <TextLineComponent tooltip="" label={`[${i}]`} value="" color="grey"></TextLineComponent>

```

Comment:
Is that the expected label? the index inside []?

Reply:
Yep it's part of the design: 
![image](https://user-images.githubusercontent.com/67929198/142431430-182535e0-8695-4e1d-9aa5-b4fd03b79fbe.png)